R/load_pbp.R

```diff
@@ -1,78 +1,8 @@
-#' Load complete nflfastR Play by Play Datasets
-#'
-#' @description Loads multiple seasons from the nflfastR datarepo either into memory
-#' or writes them into a database using some forwarded arguments in the dots.
-#'
-#' @param seasons A vector of 4-digit years associated with given NFL seasons.
-#' @param ... Additional arguments passed to an underlying function that writes
-#' the season data into a database (used by [update_db()]).
-#' @param qs Wheter to use the function [qs::qdeserialize()] for more efficient loading.
-#' @return The complete nflfastR dataset as returned by [build_nflfastR_pbp()] for
-#' all given `seasons`
-#'
-#' @seealso For information on parallel processing and progress updates please
-#' see [nflfastR].
+#' @inherit nflreadr::load_pbp
 #' @examples
 #' \donttest{
 #' pbp <- load_pbp(2019:2020)
 #' dplyr::glimpse(pbp)
 #' }
 #' @export
-load_pbp <- function(seasons, ..., qs = FALSE) {
-  dots <- rlang::dots_list(...)
-
-  if (all(c("dbConnection", "tablename") %in% names(dots))) in_db <- TRUE else in_db <- FALSE
-
-  if (isTRUE(qs) && !is_installed("qs")) {
-    cli::cli_abort("Package {.val qs} required for argument {.val qs = TRUE}. Please install it.")
-  }
-
-  most_recent <- most_recent_season()
-
-  if (!all(seasons %in% 1999:most_recent)) {
-    cli::cli_abort("Please pass valid seasons between 1999 and {most_recent}")
-  }
-
-  if (length(seasons) > 1 && is_sequential() && isFALSE(in_db)) {
-    cli::cli_alert_info(c(
-      "It is recommended to use parallel processing when trying to load multiple seasons.\n",
-      "Please consider running {.code future::plan(\"multisession\")}! ",
-      "Will go on sequentially..."
-    ))
-  }
-
-  p <- progressr::progressor(along = seasons)
-
-  if (isFALSE(in_db)) {
-    out <- furrr::future_map_dfr(seasons, single_season, p = p, qs = qs)
-  }
-
-  if (isTRUE(in_db)) {
-    purrr::walk(seasons, single_season, p, ..., qs = qs)
-    out <- NULL
-  }
-
-  return(out)
-}
-
-# Helper function that is called by load_pbp above
-single_season <- function(season, p, dbConnection = NULL, tablename = NULL, qs = FALSE) {
-  if (isTRUE(qs)) {
-    .url <- glue::glue("https://github.com/nflverse/nflfastR-data/blob/master/data/play_by_play_{season}.qs?raw=true")
-    pbp <- qs_from_url(.url)
-  }
-  if (isFALSE(qs)) {
-    .url <- glue::glue("https://github.com/nflverse/nflfastR-data/blob/master/data/play_by_play_{season}.rds?raw=true")
-    con <- url(.url)
-    pbp <- readRDS(con)
-    close(con)
-  }
-  if (!is.null(dbConnection) && !is.null(tablename)) {
-    DBI::dbWriteTable(dbConnection, tablename, pbp, append = TRUE)
-    out <- NULL
-  } else {
-    out <- pbp
-  }
-  p(sprintf("season=%g", season))
-  return(out)
-}
+load_pbp <- nflreadr::load_pbp
```

Comment:
We can probably do something with the lifecycle package that's already imported here, e.g. https://lifecycle.r-lib.org/articles/communicate.html#deprecating-an-argument-providing-a-new-default

```
@inherit nflreadr::load_pbp
#' @param qs deprecated as of v4.3.0, please use `file_type` instead
#' @examples
#' \donttest{
#' pbp <- load_pbp(2019:2020)
#' dplyr::glimpse(pbp)
#' }
#' @export
load_pbp <- function(..., qs = deprecated()){

  if (lifecycle::is_present(qs)) {
    lifecycle::deprecate_warn(
      when = "4.3.0", 
      what = "load_pbp(qs)",
      details = "The `qs` argument is deprecated and replaced by `file_type` - see `?nflreadr::load_pbp` for details."
    )
  }

  nflreadr::load_pbp(...)
}
```

Reply:
and then in the next version we can formally kill it off with the simplified re-export

Reply:
That was my initial plan but we can't inherit anything from nflreadr if the function params change. The only way this would work is to copy the function params of `nflreadr::load_pbp`. 

I tried this

```r
#' @inherit nflreadr::load_pbp
#' @param qs `r lifecycle::badge("deprecated")` has no effect and will be
#' removed in a future release.
#' @param ... bla
#' @examples
#' \donttest{
#' pbp <- load_pbp(2019:2020)
#' dplyr::glimpse(pbp)
#' }
#' @export
load_pbp <- function(..., qs = lifecycle::deprecated()){
  if (lifecycle::is_present(qs)) {
    lifecycle::deprecate_warn(
      when = "4.3.0",
      what = "load_pbp(qs = )",
      details = cli::cli_text("The {.val qs} argument is deprecated and replaced by {.val file_type} - see {.code ?nflreadr::load_pbp} for details.")
    )
  }
  nflreadr::load_pbp(...)
}
```

And this is the resulting function doc 
![grafik](https://user-images.githubusercontent.com/38586519/129759190-a4c6ed2d-f034-492a-aee2-5cca9ce1f7e7.png)

Reply:
Maybe try @inheritDotParams also?
![image](https://user-images.githubusercontent.com/38083823/129762709-23b4045b-2d2a-4f62-84ec-c70d2244c05f.png)

Reply:
no idea why lifecycle doesn't work for me but it works for you so 🤷‍♂️

Reply:
That's a great call. I think that's good enough to nicely deprecate the `qs` argument